File: app/src/main/java/com/example/anton/sprintcalendar/DefaultHolidayProvider.java
package com.example.anton.sprintcalendar;

import org.joda.time.DateTimeComparator;
import org.joda.time.LocalDate;

import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.Date;
import java.util.List;

public class DefaultHolidayProvider implements HolidayProvider {

    private static final DateTimeComparator dateComparator = DateTimeComparator.getDateOnlyInstance();
    private List<LocalDate> holidays;

    public DefaultHolidayProvider(LocalDate... holidaysList) {
        holidays = Arrays.asList(holidaysList);
    }

    @Override
    public boolean isHoliday(LocalDate date) {
        return Collections.binarySearch(holidays, date) >= 0;
    }
}

File: app/src/main/kotlin/net/sourcefusion/agiletools/sprintcalendar/DefaultHolidayProvider.kt
package net.sourcefusion.agiletools.sprintcalendar

import org.joda.time.LocalDate

import org.joda.time.DateTimeConstants.SATURDAY
import org.joda.time.DateTimeConstants.SUNDAY

class DefaultHolidayProvider(vararg val holidaysList: LocalDate) : HolidayProvider {

    override fun isHoliday(date: LocalDate): Boolean {
        return isWeekend(date) || date in holidaysList
    }

    override fun isWeekend(date: LocalDate): Boolean {
        return date.dayOfWeek == SATURDAY || date.dayOfWeek == SUNDAY
    }

    override fun isWorkingDay(localDate: LocalDate): Boolean {
        return !isHoliday(localDate)
    }


}
